.github/workflows/test.yml

```diff
@@ -102,58 +103,98 @@ jobs:
+        echo "DOCKER_BASE=$DOCKER_USER/${{ env.DOCKER_REPO }}" >> $GITHUB_ENV
         echo "DOCKER_BUILD_PLATFORMS=${DOCKER_PLATFORMS// /,}" >> $GITHUB_ENV
-    - name: Build and push
+    - name: Build and push to local registry
       uses: docker/build-push-action@v2
       with:
         context: .
         file: ./cmd/shfmt/Dockerfile
         platforms: ${{ env.DOCKER_BUILD_PLATFORMS }}
         push: true
-        tags: ${{ env.DOCKER_BASE }}:${{ env.TAG }}
-    - name: Build and push (alpine)
+        tags: localhost:5000/${{ env.DOCKER_BASE }}:${{ env.TAG }}
+    - name: Build and push to local registry (alpine)
       uses: docker/build-push-action@v2
       with:
         context: .
         file: ./cmd/shfmt/Dockerfile
         platforms: ${{ env.DOCKER_BUILD_PLATFORMS }}
         push: true
-        tags: ${{ env.DOCKER_BASE }}:${{ env.TAG }}-alpine
+        tags: localhost:5000/${{ env.DOCKER_BASE }}:${{ env.TAG }}-alpine
         target: alpine
-    - name: Test multi-architecture Docker images
+    - name: Test multi-arch Docker images locally
       run: |
-        # Enable docker daemon experimental support (for 'pull --platform').
-        config='/etc/docker/daemon.json'
-        if [[ -e "$config" ]]; then
-          sudo sed -i -e 's/{/{ "experimental": true, /' "$config"
-        else
-          echo '{ "experimental": true }' | sudo tee "$config"
-        fi
-        sudo systemctl restart docker
-        printf '%s\n' "#!/bin/sh" "echo 'hello world'" >myscript
         for platform in $DOCKER_PLATFORMS; do
           for ext in '' '-alpine'; do
-            image="${DOCKER_BASE}:${TAG}${ext}"
+            image="localhost:5000/${DOCKER_BASE}:${TAG}${ext}"
             msg="Testing docker image $image on platform $platform"
             line="${msg//?/=}"
             printf "\n${line}\n${msg}\n${line}\n"
             docker pull -q --platform "$platform" "$image"
-            if [ -n "$ext" ]; then
+            if [[ -n "$ext" ]]; then
               echo -n "Image architecture: "
               docker run --rm --entrypoint /bin/sh "$image" -c 'uname -m'
             fi
             version=$(docker run --rm "$image" --version)
             echo "shfmt version: $version"
-            if [ "$TAG" != 'latest' -a "$TAG" != "$version" ]; then
+            if [[ $TAG != "latest" ]] &&
+            [[ $TAG != "$version" ]] &&
+            ! echo "$version" | grep -q "$TAG"; then
               echo "Version mismatch: shfmt $version tagged as $TAG"
               exit 1
             fi
-            docker run --rm -v "$PWD:/mnt" -w '/mnt' "$image" -d myscript
+            docker run --rm -v "$PWD:/mnt" -w '/mnt' "$image" -d cmd/shfmt/docker-entrypoint.sh
           done
         done
+    - name: Check GitHub settings
+      if: >
+        github.event_name == 'push' &&
+        (github.ref == 'refs/heads/master' ||
+        startsWith(github.ref, 'refs/tags/')) &&
+        github.repository == 'mvdan/sh'
```

Comment:
I'd suggest refactoring this in one of 2 ways:
 1. Like in the link you sent, where they use the complex condition to set 1 environment variable (e.g. `DO_DEPLOY`) and later just check that one variable
 2. By using the more sophisticated YAML reference syntax.
```
- name: Check GitHub settings
  if: &deploy_condition >
    github.event_name == 'push' &&
    (github.ref == 'refs/heads/master' ||
    startsWith(github.ref, 'refs/tags/')) &&
    github.repository == 'mvdan/sh'
  run: |
    ... as you've got ...
- name: Login to DockerHub
  if: *deploy_condition
  uses: ... as you've got ...
```
FYI: `yaml2js` is a useful tool to check if you've written what you meant.
```
alias yaml2js='python -c '\''import sys, yaml, json; json.dump(yaml.load(sys.stdin), sys.stdout, indent=4)'\'''
```

Reply:
Thanks for the tip I'll take a look!

Regarding Yaml anchors, I liked it a lot, but I seems that it's not supported in GitHub Actions at the moment: https://github.community/t/support-for-yaml-anchors/16128/48.

I guess I'll try to implement 1.

Reply:
I assume this was done with DOCKER_DEPLOY_IMAGES, marking as resolved.

I would personally stay away from yaml anchors too, because they're generally not well understood or supported.